packages/tupaia-web/src/features/ExportSettings/ExportSettingsContext.tsx

```diff
@@ -0,0 +1,130 @@
+  exportWithLabels: false,
+  exportWithTable: true,
+  exportWithTableDisabled: false,
+} as ExportSettingsReducerState;
+
+// This is the context for the export settings
+export const ExportSettingsContext = createContext(defaultContext);
+
+export enum ExportSettingsActionTypes {
+  SET_EXPORT_FORMAT = 'SET_EXPORT_FORMAT',
+  SET_EXPORT_WITH_LABELS = 'SET_EXPORT_WITH_LABELS',
+  SET_EXPORT_WITH_TABLE = 'SET_EXPORT_WITH_TABLE',
+  RESET_EXPORT_STATE = 'RESET_EXPORT_STATE',
+}
+
+interface ActionT {
+  type: ExportSettingsActionTypes;
+  payload?: boolean | ExportFormats | string | null;
+}
+
+export const ExportSettingsDispatchContext = createContext<Dispatch<ActionT> | null>(null);
+
+export const exportSettingsReducer = (
+  state: ExportSettingsReducerState,
+  action: ActionT,
+): ExportSettingsReducerState => {
+  switch (action.type) {
+    case ExportSettingsActionTypes.SET_EXPORT_FORMAT:
+      return {
+        ...state,
+        exportFormat: action.payload as ExportFormats,
+      };
+    case ExportSettingsActionTypes.SET_EXPORT_WITH_LABELS:
+      return {
+        ...state,
+        exportWithLabels: action.payload as boolean,
+      };
+    case ExportSettingsActionTypes.SET_EXPORT_WITH_TABLE:
+      return {
+        ...state,
+        exportWithTable: action.payload as boolean,
+      };
+    case ExportSettingsActionTypes.RESET_EXPORT_STATE:
+      return {
+        ...state,
+        exportFormat: action.payload === 'matrix' ? ExportFormats.XLSX : ExportFormats.PNG,
+        exportWithLabels: false,
+        exportWithTable: true,
+        exportWithTableDisabled: false,
+      };
+    default:
+      return state;
+  }
+};
+
+export const useExportSettings = () => {
+  const { exportFormat, exportWithLabels, exportWithTable, exportWithTableDisabled } =
+    useContext(ExportSettingsContext);
+  const dispatch = useContext(ExportSettingsDispatchContext);
+
+  const setExportFormat = (e: ChangeEvent<HTMLInputElement>) =>
+    dispatch?.({
+      type: ExportSettingsActionTypes.SET_EXPORT_FORMAT,
+      payload: e.target.value,
+    });
+
+  const setExportWithLabels = (e: ChangeEvent<HTMLInputElement>) => {
+    dispatch?.({
+      type: ExportSettingsActionTypes.SET_EXPORT_WITH_LABELS,
+      payload: e.target.checked,
+    });
+  };
+
+  const setExportWithTable = (e: ChangeEvent<HTMLInputElement>) => {
+    dispatch?.({
+      type: ExportSettingsActionTypes.SET_EXPORT_WITH_TABLE,
+      payload: e.target.checked,
+    });
+  };
+
+  return {
+    exportFormat,
+    exportWithLabels,
+    exportWithTable,
+    exportWithTableDisabled,
+    setExportFormat,
+    setExportWithLabels,
+    setExportWithTable,
+  };
+};
+
+export const ExportSettingsContextProvider = ({
+  defaultSettings,
+  children,
+}: {
+  defaultSettings?: ExportSettingsReducerState;
+  children: React.ReactNode;
+}) => {
+  const [state, dispatch] = useReducer(exportSettingsReducer, {
+    ...defaultContext,
+    ...(defaultSettings || {}),
+  });
```

Comment:
Yeah for sure, we can do that!

Reply:
@rohan-bes I think perhaps a reducer in this case is useful, because we have a utility hook for `useExportDashboardItem` which makes use of the export settings context (and reducer). When there is state within a hook, everytime you make use of that hook it will sort of create a new instance of that hook and it's state, which means we can't actually share data between components as we want to here. SO in this case, because there are multiple sub components accessing the state, I think a reducer is our best option